### PR TITLE
Fix Schema.

### PR DESCRIPTION
**Pull request by wangbwn**

### CODE REVIEW

**Comment by wangbwn** [comment]
[External Code]	
>	[Exception] Azure.ResourceManager.DataFactory.dll!Azure.ResourceManager.DataFactory.Models.AvroDataset.Azure.Core.IUtf8JsonSerializable.Write(System.Text.Json.Utf8JsonWriter writer) Line 33	C#
 	[Exception] Azure.ResourceManager.DataFactory.dll!Azure.Core.Utf8JsonWriterExtensions.WriteObjectValue(System.Text.Json.Utf8JsonWriter writer, object value) Line 79	C#
 	[Exception] Azure.ResourceManager.DataFactory.dll!Azure.ResourceManager.DataFactory.DataFactoryDatasetData.Azure.Core.IUtf8JsonSerializable.Write(System.Text.Json.Utf8JsonWriter writer) Line 22	C#
 	[Exception] Azure.ResourceManager.DataFactory.dll!Azure.Core.Utf8JsonWriterExtensions.WriteObjectValue(System.Text.Json.Utf8JsonWriter writer, object value) Line 79	C#
 	[Exception] Azure.ResourceManager.DataFactory.dll!Azure.ResourceManager.DataFactory.DatasetsRestOperations.CreateCreateOrUpdateRequest(string subscriptionId, string resourceGroupName, string factoryName, string datasetName, Azure.ResourceManager.DataFactory.DataFactoryDatasetData data, string ifMatch) Line 143	C#
 	[Exception] Azure.ResourceManager.DataFactory.dll!Azure.ResourceManager.DataFactory.DatasetsRestOperations.CreateOrUpdate(string subscriptionId, string resourceGroupName, string factoryName, string datasetName, Azure.ResourceManager.DataFactory.DataFactoryDatasetData data, string ifMatch, System.Threading.CancellationToken cancellationToken) Line 201	C#
 	Azure.ResourceManager.DataFactory.dll!Azure.ResourceManager.DataFactory.DataFactoryDatasetCollection.CreateOrUpdate(Azure.WaitUntil waitUntil, string datasetName, Azure.ResourceManager.DataFactory.DataFactoryDatasetData data, string ifMatch, System.Threading.CancellationToken cancellationToken) Line 136	C#
 	[External Code]	
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.DiagnosticScopeValidatingInterceptor.WrapAsyncResultCore.__Await|0() Line 93	C#
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.InstrumentResultInterceptor.InstrumentOperationInterceptor<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.DataFactory.DataFactoryDatasetResource>>(Castle.DynamicProxy.IInvocation invocation, System.Func<System.Threading.Tasks.ValueTask<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.DataFactory.DataFactoryDatasetResource>>> innerTask) Line 105	C#
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.DiagnosticScopeValidatingInterceptor.WrapAsyncResultCore<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.DataFactory.DataFactoryDatasetResource>>(Castle.DynamicProxy.IInvocation invocation, System.Type genericType, Azure.Core.TestFramework.AsyncCallInterceptor<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.DataFactory.DataFactoryDatasetResource>> wrap) Line 126	C#
 	[External Code]	
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.DiagnosticScopeValidatingInterceptor.WrapAsyncResult(Castle.DynamicProxy.IInvocation invocation, object target, System.Reflection.MethodInfo interceptorMethod) Line 73	C#
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.InstrumentResultInterceptor.Intercept(Castle.DynamicProxy.IInvocation invocation) Line 77	C#
 	[External Code]	
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.DiagnosticScopeValidatingInterceptor.WrapAsyncResultCore.__Await|0() Line 93	C#
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.DiagnosticScopeValidatingInterceptor.ValidateDiagnosticScopeInterceptor.AnonymousMethod__0() Line 174	C#
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.DiagnosticScopeValidatingInterceptor.ValidateDiagnosticScope<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.DataFactory.DataFactoryDatasetResource>>(System.Func<System.Threading.Tasks.ValueTask<(Azure.ResourceManager.ArmOperation<Azure.ResourceManager.DataFactory.DataFactoryDatasetResource> Result, bool SkipChecks)>> action, System.Reflection.MethodInfo methodInfo, string source) Line 208	C#
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.DiagnosticScopeValidatingInterceptor.ValidateDiagnosticScopeInterceptor<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.DataFactory.DataFactoryDatasetResource>>(Castle.DynamicProxy.IInvocation invocation, System.Func<System.Threading.Tasks.ValueTask<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.DataFactory.DataFactoryDatasetResource>>> innerTask) Line 174	C#
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.DiagnosticScopeValidatingInterceptor.WrapAsyncResultCore<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.DataFactory.DataFactoryDatasetResource>>(Castle.DynamicProxy.IInvocation invocation, System.Type genericType, Azure.Core.TestFramework.AsyncCallInterceptor<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.DataFactory.DataFactoryDatasetResource>> wrap) Line 126	C#
 	[External Code]	
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.DiagnosticScopeValidatingInterceptor.WrapAsyncResult(Castle.DynamicProxy.IInvocation invocation, object target, System.Reflection.MethodInfo interceptorMethod) Line 73	C#
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.DiagnosticScopeValidatingInterceptor.Intercept(Castle.DynamicProxy.IInvocation invocation) Line 45	C#
 	[External Code]	
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.GetOriginalInterceptor.Intercept(Castle.DynamicProxy.IInvocation invocation) Line 28	C#
 	[External Code]	
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.ManagementInterceptor.Intercept(Castle.DynamicProxy.IInvocation invocation) Line 27	C#
 	[External Code]	
 	Azure.ResourceManager.DataFactory.Tests.dll!Azure.ResourceManager.DataFactory.Tests.Scenario.DataFactoryDatasetResourceTests.Dataset_Avro_Create() Line 2255	C#
 	[External Code]	
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.RecordedTestAttribute.RecordedTestAttributeCommand.Execute(NUnit.Framework.Internal.TestExecutionContext context) Line 49	C#
 	[External Code]

File: sdk/datafactory/Azure.ResourceManager.DataFactory/src/Customized/AvroDataset.cs [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Text.Json;
+using Azure.Core;
+using Azure.Core.Expressions.DataFactory;
+
+namespace Azure.ResourceManager.DataFactory.Models
```
Comment (archerzz) [inline sdk/datafactory/Azure.ResourceManager.DataFactory/src/Customized/AvroDataset.cs:13]: Better to split this file into `AvroDataset.cs` and `AvroDataset.Serliaization.cs`.

File: sdk/datafactory/Azure.ResourceManager.DataFactory/src/Customized/AvroDataset.cs [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        {
+            DataLocation = dataLocation;
+            AvroCompressionCodec = avroCompressionCodec;
+            AvroCompressionLevel = avroCompressionLevel;
+            DatasetType = datasetType ?? "Avro";
+            Schema = schema;
+        }
+
+        /// <summary> Columns that define the physical type schema of the dataset. Type: array (or Expression with resultType array), itemType: DatasetSchemaDataElement. </summary>
+        public new DataFactoryElement<BinaryData> Schema { get; set; }
```
Comment (archerzz) [inline sdk/datafactory/Azure.ResourceManager.DataFactory/src/Customized/AvroDataset.cs:48]: You didn't override the parent `Schema` property. Are you sure this is what you want? https://github.com/Azure/azure-sdk-for-net/blob/9ac52a165101b727ff606dd50dd82f771ec7c710/sdk/datafactory/Azure.ResourceManager.DataFactory/src/Generated/Models/DataFactoryDatasetProperties.cs#L65

File: sdk/datafactory/Azure.ResourceManager.DataFactory/src/Customized/AvroDataset.cs [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        continue;
+                    }
+                    schema = JsonSerializer.Deserialize<DataFactoryElement<BinaryData>>(property.Value.GetRawText());
+                    continue;
+                }
+                if (property.NameEquals("linkedServiceName"u8))
+                {
+                    linkedServiceName = JsonSerializer.Deserialize<Core.Expressions.DataFactory.DataFactoryLinkedServiceReference>(property.Value.GetRawText());
```
Comment (archerzz) [inline sdk/datafactory/Azure.ResourceManager.DataFactory/src/Customized/AvroDataset.cs:101]:
```suggestion
                    linkedServiceName = JsonSerializer.Deserialize<DataFactoryLinkedServiceReference>(property.Value.GetRawText());
```

**Review by archerzz — CHANGES_REQUESTED** [review]
Please double check the override of `Schema` property.

Also fix the pipeline error.

**Comment by ruowan** [comment]
```
 	[External Code]	
>	[Exception] Azure.ResourceManager.DataFactory.dll!Azure.ResourceManager.DataFactory.Models.AvroDataset.Azure.Core.IUtf8JsonSerializable.Write(System.Text.Json.Utf8JsonWriter writer) Line 33	C#
 	[Exception] Azure.ResourceManager.DataFactory.dll!Azure.Core.Utf8JsonWriterExtensions.WriteObjectValue(System.Text.Json.Utf8JsonWriter writer, object value) Line 79	C#
 	[Exception] Azure.ResourceManager.DataFactory.dll!Azure.ResourceManager.DataFactory.DataFactoryDatasetData.Azure.Core.IUtf8JsonSerializable.Write(System.Text.Json.Utf8JsonWriter writer) Line 22	C#
 	[Exception] Azure.ResourceManager.DataFactory.dll!Azure.Core.Utf8JsonWriterExtensions.WriteObjectValue(System.Text.Json.Utf8JsonWriter writer, object value) Line 79	C#
 	[Exception] Azure.ResourceManager.DataFactory.dll!Azure.ResourceManager.DataFactory.DatasetsRestOperations.CreateCreateOrUpdateRequest(string subscriptionId, string resourceGroupName, string factoryName, string datasetName, Azure.ResourceManager.DataFactory.DataFactoryDatasetData data, string ifMatch) Line 143	C#
 	[Exception] Azure.ResourceManager.DataFactory.dll!Azure.ResourceManager.DataFactory.DatasetsRestOperations.CreateOrUpdate(string subscriptionId, string resourceGroupName, string factoryName, string datasetName, Azure.ResourceManager.DataFactory.DataFactoryDatasetData data, string ifMatch, System.Threading.CancellationToken cancellationToken) Line 201	C#
 	Azure.ResourceManager.DataFactory.dll!Azure.ResourceManager.DataFactory.DataFactoryDatasetCollection.CreateOrUpdate(Azure.WaitUntil waitUntil, string datasetName, Azure.ResourceManager.DataFactory.DataFactoryDatasetData data, string ifMatch, System.Threading.CancellationToken cancellationToken) Line 136	C#
 	[External Code]	
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.DiagnosticScopeValidatingInterceptor.WrapAsyncResultCore.__Await|0() Line 93	C#
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.InstrumentResultInterceptor.InstrumentOperationInterceptor<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.DataFactory.DataFactoryDatasetResource>>(Castle.DynamicProxy.IInvocation invocation, System.Func<System.Threading.Tasks.ValueTask<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.DataFactory.DataFactoryDatasetResource>>> innerTask) Line 105	C#
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.DiagnosticScopeValidatingInterceptor.WrapAsyncResultCore<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.DataFactory.DataFactoryDatasetResource>>(Castle.DynamicProxy.IInvocation invocation, System.Type genericType, Azure.Core.TestFramework.AsyncCallInterceptor<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.DataFactory.DataFactoryDatasetResource>> wrap) Line 126	C#
 	[External Code]	
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.DiagnosticScopeValidatingInterceptor.WrapAsyncResult(Castle.DynamicProxy.IInvocation invocation, object target, System.Reflection.MethodInfo interceptorMethod) Line 73	C#
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.InstrumentResultInterceptor.Intercept(Castle.DynamicProxy.IInvocation invocation) Line 77	C#
 	[External Code]	
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.DiagnosticScopeValidatingInterceptor.WrapAsyncResultCore.__Await|0() Line 93	C#
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.DiagnosticScopeValidatingInterceptor.ValidateDiagnosticScopeInterceptor.AnonymousMethod__0() Line 174	C#
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.DiagnosticScopeValidatingInterceptor.ValidateDiagnosticScope<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.DataFactory.DataFactoryDatasetResource>>(System.Func<System.Threading.Tasks.ValueTask<(Azure.ResourceManager.ArmOperation<Azure.ResourceManager.DataFactory.DataFactoryDatasetResource> Result, bool SkipChecks)>> action, System.Reflection.MethodInfo methodInfo, string source) Line 208	C#
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.DiagnosticScopeValidatingInterceptor.ValidateDiagnosticScopeInterceptor<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.DataFactory.DataFactoryDatasetResource>>(Castle.DynamicProxy.IInvocation invocation, System.Func<System.Threading.Tasks.ValueTask<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.DataFactory.DataFactoryDatasetResource>>> innerTask) Line 174	C#
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.DiagnosticScopeValidatingInterceptor.WrapAsyncResultCore<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.DataFactory.DataFactoryDatasetResource>>(Castle.DynamicProxy.IInvocation invocation, System.Type genericType, Azure.Core.TestFramework.AsyncCallInterceptor<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.DataFactory.DataFactoryDatasetResource>> wrap) Line 126	C#
 	[External Code]	
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.DiagnosticScopeValidatingInterceptor.WrapAsyncResult(Castle.DynamicProxy.IInvocation invocation, object target, System.Reflection.MethodInfo interceptorMethod) Line 73	C#
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.DiagnosticScopeValidatingInterceptor.Intercept(Castle.DynamicProxy.IInvocation invocation) Line 45	C#
 	[External Code]	
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.GetOriginalInterceptor.Intercept(Castle.DynamicProxy.IInvocation invocation) Line 28	C#
 	[External Code]	
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.ManagementInterceptor.Intercept(Castle.DynamicProxy.IInvocation invocation) Line 27	C#
 	[External Code]	
 	Azure.ResourceManager.DataFactory.Tests.dll!Azure.ResourceManager.DataFactory.Tests.Scenario.DataFactoryDatasetResourceTests.Dataset_Avro_Create() Line 2255	C#
 	[External Code]	
 	Azure.Core.TestFramework.dll!Azure.Core.TestFramework.RecordedTestAttribute.RecordedTestAttributeCommand.Execute(NUnit.Framework.Internal.TestExecutionContext context) Line 49	C#
 	[External Code]	
```

**Pull request closed without merging**